certbot/src/certbot/_internal/client.py

```diff
@@ -409,7 +418,7 @@ def obtain_certificate(self, domains: List[str], old_keypath: Optional[str] = No
             )
             csr = util.CSR(file=None, form="pem",
                            data=acme_crypto_util.make_csr(
-                               key.pem, domains, self.config.must_staple))
+                               key.pem, dns_domains, self.config.must_staple, ipaddrs=ip_addrs))
```

Comment:
dns_domains as a name is a little redundant. But if you make the change I suggest above, I think you will naturally wind up reverting this name change.